src/hotspot/share/gc/shenandoah/shenandoahOldGeneration.hpp

```diff
@@ -0,0 +1,22 @@
+
+#ifndef SHARE_VM_GC_SHENANDOAH_SHENANDOAHOLDGENERATION_HPP
+#define SHARE_VM_GC_SHENANDOAH_SHENANDOAHOLDGENERATION_HPP
+
+#include "gc/shenandoah/shenandoahGeneration.hpp"
+
```

Comment:
You should add a forward declaration of ShenandoahHeapRegion and ShenandoahHeapRegionClosure.